### YamlMime:JavaType
uid: "com.azure.resourcemanager.compute.models.ApiErrorException"
fullName: "com.azure.resourcemanager.compute.models.ApiErrorException"
name: "ApiErrorException"
nameWithType: "ApiErrorException"
summary: "Exception thrown for an invalid response with Api<wbr>Error information."
inheritances:
- "<xref href=\"java.lang.Object?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Throwable?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Exception?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.RuntimeException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.exception.AzureException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.exception.HttpResponseException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.management.exception.ManagementException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
inheritedClassMethods:
- classRef: "<xref href=\"com.azure.core.exception.HttpResponseException?alt=com.azure.core.exception.HttpResponseException&text=HttpResponseException\" data-throw-if-not-resolved=\"False\" />"
  methodsRef:
  - "<xref href=\"com.azure.core.exception.HttpResponseException.getResponse()?alt=com.azure.core.exception.HttpResponseException.getResponse&text=getResponse\" data-throw-if-not-resolved=\"False\" />"
- classRef: "<xref href=\"com.azure.core.management.exception.ManagementException?alt=com.azure.core.management.exception.ManagementException&text=ManagementException\" data-throw-if-not-resolved=\"False\" />"
  methodsRef:
  - "<xref href=\"com.azure.core.management.exception.ManagementException.getValue()?alt=com.azure.core.management.exception.ManagementException.getValue&text=getValue\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.management.exception.ManagementException.toString()?alt=com.azure.core.management.exception.ManagementException.toString&text=toString\" data-throw-if-not-resolved=\"False\" />"
- classRef: "java.lang.<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html\">Object</a>"
  methodsRef:
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#clone--\">clone</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#equals-java.lang.Object-\">equals</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#finalize--\">finalize</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#getClass--\">getClass</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode--\">hashCode</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notify--\">notify</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notifyAll--\">notifyAll</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait--\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-int-\">wait</a>"
- classRef: "java.lang.<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html\">Throwable</a>"
  methodsRef:
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#addSuppressed-java.lang.Throwable-\">addSuppressed</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#fillInStackTrace--\">fillInStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getCause--\">getCause</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getLocalizedMessage--\">getLocalizedMessage</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getMessage--\">getMessage</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getStackTrace--\">getStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getSuppressed--\">getSuppressed</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#initCause-java.lang.Throwable-\">initCause</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace--\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace-java.io.PrintStream-\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace-java.io.PrintWriter-\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#setStackTrace-java.lang.StackTraceElement[]-\">setStackTrace</a>"
syntax: "public final class ApiErrorException extends ManagementException"
constructors:
- uid: "com.azure.resourcemanager.compute.models.ApiErrorException.ApiErrorException(java.lang.String,com.azure.core.http.HttpResponse)"
  fullName: "com.azure.resourcemanager.compute.models.ApiErrorException.ApiErrorException(String message, HttpResponse response)"
  name: "ApiErrorException(String message, HttpResponse response)"
  nameWithType: "ApiErrorException.ApiErrorException(String message, HttpResponse response)"
  summary: "Initializes a new instance of the Api<wbr>Error<wbr>Exception class."
  parameters:
  - description: "the exception message or the response content if a message is not available."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "the HTTP response."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public ApiErrorException(String message, HttpResponse response)"
  desc: "Initializes a new instance of the ApiErrorException class."
- uid: "com.azure.resourcemanager.compute.models.ApiErrorException.ApiErrorException(java.lang.String,com.azure.core.http.HttpResponse,com.azure.resourcemanager.compute.models.ApiError)"
  fullName: "com.azure.resourcemanager.compute.models.ApiErrorException.ApiErrorException(String message, HttpResponse response, ApiError value)"
  name: "ApiErrorException(String message, HttpResponse response, ApiError value)"
  nameWithType: "ApiErrorException.ApiErrorException(String message, HttpResponse response, ApiError value)"
  summary: "Initializes a new instance of the Api<wbr>Error<wbr>Exception class."
  parameters:
  - description: "the exception message or the response content if a message is not available."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "the HTTP response."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  - description: "the deserialized response value."
    name: "value"
    type: "<xref href=\"com.azure.resourcemanager.compute.models.ApiError?alt=com.azure.resourcemanager.compute.models.ApiError&text=ApiError\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public ApiErrorException(String message, HttpResponse response, ApiError value)"
  desc: "Initializes a new instance of the ApiErrorException class."
methods:
- uid: "com.azure.resourcemanager.compute.models.ApiErrorException.getValue()"
  fullName: "com.azure.resourcemanager.compute.models.ApiErrorException.getValue()"
  name: "getValue()"
  nameWithType: "ApiErrorException.getValue()"
  overridden: "com.azure.core.management.exception.ManagementException.getValue()"
  syntax: "public ApiError getValue()"
  returns:
    type: "<xref href=\"com.azure.resourcemanager.compute.models.ApiError?alt=com.azure.resourcemanager.compute.models.ApiError&text=ApiError\" data-throw-if-not-resolved=\"False\" />"
type: "class"
desc: "Exception thrown for an invalid response with ApiError information."
metadata: {}
package: "com.azure.resourcemanager.compute.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-compute:2.19.0
